tracer.go

```diff
@@ -22,26 +23,4 @@ type Tracer interface {
 	//         SetTag("lucky_number", 42)
 	//
 	StartTrace(operationName string) Span
-
-	// Like `StartTrace`, but the returned `Span` is made a child of `parent`.
-	//
-	// The `parent` parameter can either be a `context.Context` or an
-	// `opentracing.TraceContext`. In the former case, the implementation
-	// attempts to extract an `opentracing.Span` using `SpanFromGoContext()`.
-	JoinTrace(operationName string, parent interface{}) Span
```

Comment:
this is not a rhetorical question, so apologies if it sounds snarky (it's not intended to): how much time have you spent looking at calling code for tracing systems?

My experience is that the vast majority (95%? 99%?) of it would only have to use the `Span` API we have now (and, in Go, perhaps the stuff in gocontext.go), and I like that simplicity for the user's mental model.